.github/workflows/ci.yml

```diff
@@ -236,103 +195,51 @@ jobs:
         sync-uri = https://raw.githubusercontent.com/257/binpkgs/main
         EOF
 
-    - name: Build ${{ matrix.distro }}/${{ matrix.format }}
-      run: |
-        sudo python3 -m mkosi build
-
-    - name: Boot ${{ matrix.distro }}/${{ matrix.format }}
-      if: matrix.format != 'tar' &&
-          matrix.format != 'cpio'
-      run: sudo ./tests/pexpect/boot.py python3 -m mkosi boot
-
-    - name: Build ${{ matrix.distro }}/${{ matrix.format }} UEFI UKI
-      if: matrix.format != 'directory' &&
-          matrix.format != 'plain_squashfs' &&
-          matrix.format != 'tar' &&
-          matrix.format != 'cpio'
+    - name: Build/Boot ${{ matrix.distro }}/${{ matrix.format }}
+      run: MKOSI_TEST_DEFAULT_VERB=boot sudo python3 -m pytest -m integration tests
+
+    - name: Write shared mkosi config UEFI UKI
       run: |
         tee mkosi.default <<- EOF
+        [Distribution]
+        Distribution=${{ matrix.distro }}
+
         [Output]
-        Bootable=yes
+        Format=${{ matrix.format }}
         BootProtocols=uefi
-
-        [Host]
-        QemuHeadless=yes
         EOF
 
-        sudo python3 -m mkosi --force build
-
-    - name: Boot ${{ matrix.distro }}/${{ matrix.format }} UEFI UKI
-      if: matrix.format != 'directory' &&
-          matrix.format != 'plain_squashfs' &&
-          matrix.format != 'gpt_squashfs' &&
-          matrix.format != 'tar' &&
-          matrix.format != 'cpio' &&
-          !(matrix.distro == 'ubuntu' && matrix.format == 'gpt_squashfs')
-      run: sudo ./tests/pexpect/boot.py python3 -m mkosi qemu
-
-    - name: Build ${{ matrix.distro }}/${{ matrix.format }} UEFI
-      if: matrix.format != 'directory' &&
-          matrix.format != 'plain_squashfs' &&
-          matrix.format != 'tar' &&
-          matrix.format != 'cpio' && (
-          matrix.distro == 'arch' ||
-          matrix.distro == 'centos' ||
-          matrix.distro == 'rocky' ||
-          matrix.distro == 'alma' ||
-          matrix.distro == 'fedora' ||
-          matrix.distro == 'gentoo')
+    - name: Build/Boot ${{ matrix.distro }}/${{ matrix.format }} UEFI UKI
+      run: MKOSI_TEST_DEFAULT_VERB=qemu sudo python3 -m pytest -m integration tests
+
+    - name: Write shared mkosi config UEFI
       run: |
         tee mkosi.default <<- EOF
+        [Distribution]
+        Distribution=${{ matrix.distro }}
+
         [Output]
-        Bootable=yes
+        Format=${{ matrix.format }}
```

Comment:
Yikes, nice find, turns out all these steps were running under the default format instead of the specified one